src/plugins/obstacle_avoidance_server/obstacle_avoidance_server_impl.cpp

```diff
@@ -0,0 +1,108 @@
+#include "obstacle_avoidance_server_impl.h"
+
+namespace mavsdk {
+
+ObstacleAvoidanceServerImpl::ObstacleAvoidanceServerImpl(System& system) : PluginImplBase(system)
+{
+    _parent->register_plugin(this);
+}
+
+ObstacleAvoidanceServerImpl::ObstacleAvoidanceServerImpl(std::shared_ptr<System> system) :
+    PluginImplBase(system)
+{
+    _parent->register_plugin(this);
+}
+
+ObstacleAvoidanceServerImpl::~ObstacleAvoidanceServerImpl()
+{
+    _parent->unregister_plugin(this);
+}
+
+void ObstacleAvoidanceServerImpl::init()
+{
+    using namespace std::placeholders;
+
+    _parent->register_mavlink_command_handler(
+        MAV_CMD_COMPONENT_CONTROL,
+        std::bind(&ObstacleAvoidanceServerImpl::process_component_control_command, this, _1),
+        this);
+}
+
+void ObstacleAvoidanceServerImpl::deinit()
+{
+    _parent->unregister_all_mavlink_command_handlers(this);
+}
+
+void ObstacleAvoidanceServerImpl::enable() {}
+
+void ObstacleAvoidanceServerImpl::disable() {}
+
+mavlink_message_t ObstacleAvoidanceServerImpl::process_component_control_command(
+    const MavlinkCommandReceiver::CommandLong& command)
+{
+    mavlink_message_t command_ack;
```

Comment:
In consistency with the rest of the code base I would call this message because that's what it essentially is, even though it contains a command_ack.
```suggestion
    mavlink_message_t message;
```

Reply:
Or `ack_message`?

Reply:
ack_message looks better.